### PR TITLE
Add ability for sync-vm to create a vm

### PR DESCRIPTION
**Pull request by dpsmith**

### CODE REVIEW

**Review by jandryuk — COMMENTED** [review]
The code looks good.  The commit message should explain the added command.  I was going to say the templates command should be mentioned in the commit message and/or command help, but it isn't wired up to be callable from cmd.py?

**Comment by dpsmith** [comment]
Will fix missing explanation in commit message. As for list_templates, began wiring it in until I realized that upstream rpc-proxy rules do not allow for the sync to call list_templates. Thus why it was not wired up in to cmd.py.

**Comment by dpsmith** [comment]
Commit message updated.

**Review by jandryuk — APPROVED** [review]
Thanks

File: pysynchronizer/cmd.py [hunk 1]
```diff
@@ -87,10 +87,22 @@ def do_vms(self, arg_str):
...
-        """Not Implemented"""
-        print("Not Implemented\n")
+        """Usage: create_vm {template name} [path to json]"""
+
+        tmplate, json_file = arg_str.split(None, 1)
+
+        json = ""
+        try:
+            with open(json_file, 'r') as f:
+                json = f.read()
```
Comment (jandryuk) [inline pysynchronizer/cmd.py:98]: insert:
```
except FileNotFoundError:
    json = json_file
```
So you can do
`sync-cmd xenmgr create_vm new-vm '{"name": "Win10"}'` without a temp file?

If it's a typo-ed filename, you get a Malformed JSON error:
org.freedesktop.DBus.Error.Failed: 223:IO error: malformed JSON: Malformed JSON: invalid token in this context /typo

File: pysynchronizer/cmd.py [hunk 2]
```diff
@@ -87,10 +87,22 @@ def do_vms(self, arg_str):
...
+        except OSError:
+            print('  ERR: failure reading json file %s' % json_file)
+            return
+
+        path = self.xenmgr.create_vm(tmplate, json)
+
+        print(dbus_path_to_uuid(path))
```
Comment (jandryuk) [inline pysynchronizer/cmd.py:105]:
```
sync-cmd xenmgr create_vm new-vm /tmp/json
Unexpected exception:
        name 'dbus_path_to_uuid' is not defined.
```        
missing import?

**Comment by dpsmith** [comment]
This PR will be replace with a new updated PR.

**Pull request closed without merging**